Dockerfile

```diff
@@ -19,166 +20,293 @@
     ./gradlew --no-daemon --stacktrace -Pversion=$ORT_VERSION :cli:distTar :helper-cli:startScripts
 
-FROM adoptopenjdk:11-jre-hotspot-bionic
-
-ENV \
-    # Package manager versions.
-    BOWER_VERSION=1.8.8 \
-    CARGO_VERSION=0.54.0-0ubuntu1~18.04.1 \
-    COMPOSER_VERSION=1.6.3-1 \
-    CONAN_VERSION=1.40.3 \
-    GO_DEP_VERSION=0.5.4 \
-    GO_VERSION=1.16.5 \
-    HASKELL_STACK_VERSION=2.1.3 \
-    NPM_VERSION=7.20.6 \
-    PYTHON_PIPENV_VERSION=2018.11.26 \
-    PYTHON_VIRTUALENV_VERSION=15.1.0 \
-    SBT_VERSION=1.3.8 \
-    YARN_VERSION=1.22.10 \
-    # SDK versions.
-    ANDROID_SDK_VERSION=6858069 \
-    # Installation directories.
-    ANDROID_HOME=/opt/android-sdk \
-    GOPATH=$HOME/go
-
-ENV DEBIAN_FRONTEND=noninteractive \
-    PATH="$PATH:$HOME/.local/bin:$GOPATH/bin:/opt/go/bin:$GEM_PATH/bin"
+RUN mkdir -p /opt/ort \
+    && tar xf /usr/local/src/ort/cli/build/distributions/ort-$ORT_VERSION.tar -C /opt/ort --strip-components 1 \
+    && cp -a /usr/local/src/ort/scripts/*.sh /opt/ort/bin/ \
+    && cp -a /usr/local/src/ort/helper-cli/build/scripts/orth /opt/ort/bin/ \
+    && cp -a /usr/local/src/ort/helper-cli/build/libs/helper-cli-*.jar /opt/ort/lib/
+
+#------------------------------------------------------------------------
+# PYTHON - Build python as a separate component with pyenv
+FROM build as pythonbuild
+
+ARG CONAN_VERSION=1.40.3
+ARG PYTHON_VERSION=3.6.15
+ARG PYTHON_VIRTUALENV_VERSION=15.1.0
+ARG PIPTOOL_VERSION=21.3.1
+
+ENV PYENV_ROOT=/opt/python
+RUN curl -L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash
+ENV PATH=/opt/python/bin:$PATH
+# Python 3.6.x series has problems with alignment with modern gcc
+# As we don not want patch python versions we use clang as compiler
+RUN CC=clang-9 CXX=clang++9 pyenv install "${PYTHON_VERSION}"
+RUN pyenv global "${PYTHON_VERSION}"
+ENV PATH=/opt/python/shims:$PATH
+RUN pip install -U \
+    conan=="${CONAN_VERSION}" \
+    pip=="${PIPTOOL_VERSION}" \
+    pipenv \
+    Mercurial \
+    virtualenv=="${PYTHON_VIRTUALENV_VERSION}" \
+    wheel
+
+COPY docker/python.sh /etc/ort/bash_modules
+
+#------------------------------------------------------------------------
+# RUBY - Build ruby as a separate component with rbenv
+FROM build as rubybuild
+
+ARG RUBY_VERSION=2.7.4
+ENV RBENV_ROOT=/opt/rbenv
+ENV PATH=${RBENV_ROOT}/bin:${RBENV_ROOT}/shims/:${RBENV_ROOT}/plugins/ruby-build/bin:$PATH
+
+RUN git clone --depth 1 https://github.com/rbenv/rbenv.git ${RBENV_ROOT}
+RUN git clone --depth 1 https://github.com/rbenv/ruby-build.git "$(rbenv root)"/plugins/ruby-build
+WORKDIR ${RBENV_ROOT}
+RUN src/configure \
+    && make -C src
+RUN rbenv install ${RUBY_VERSION} \
+    && rbenv global ${RUBY_VERSION} \
+    && gem install bundler cocoapods
+
+# Install 'CocoaPods'. As https://github.com/CocoaPods/CocoaPods/pull/10609 is needed but not yet released.
```

Comment:
https://github.com/CocoaPods/CocoaPods/pull/10609 was merged and I believe also released?

Reply:
Maybe, but i do not want change this since is this way on upstream Dockerfile. If we merge, then in later step i change to newer version of CocoaPods.

Reply:
Please see again how this is now done in the "legacy" `Dockerfile`.